src/high-level/lapack-templates.lisp

```diff
@@ -94,9 +105,29 @@
                 1 ;; NOTE: This corresponds to the stride of X
                 beta
                 (storage target)
-                1 ;; NOTE: THis corresponds to the stride of TARGET
+                1 ;; NOTE: This corresponds to the stride of TARGET
+                )
+               target)))))
+
+     (defmethod mult ((a ,vector-class) (b ,vector-class) &key target (alpha ,(coerce 1 type)) (beta ,(coerce 0 type)) transa transb)
```

Comment:
Just thinking... It probably does make sense to define multiplication of two vectors as just the dot product but we need a way of representing the outer product too. Probably out of the scope of this PR but wanted to mention

Reply:
It turns out to be pretty straightforward to distinguish row vectors and column vectors as distinct structures; then multiplying `(@ (transpose x) y)` gives you inner product, and `(@x (transpose y))` gives you outer product; all without giving up specialized matrix-vector and vector-vector BLAS calls.  (This would also address @notmgsk's concerns about losing type checking, fwiw).

I've got a draft implementation of this idea on another branch that I can easily merge to this one if there's any appetite for it; it includes a no-copying transpose method for vectors, since otherwise this scheme would presumably be needlessly slow.